### PR TITLE
Add the jsfContainer-2.3 feature.

### PR DESCRIPTION
**Pull request by hibell**

### CODE REVIEW

**Comment by hibell** [comment]
fixes #1970

**Comment by hibell** [comment]
#build

**Comment by bcpswope** [comment]
Are you just moving files over? Need ID review?

**Comment by hibell** [comment]
@bcpswope The only file that I found that appears to be relevant is *com.ibm.websphere.appserver.jsfContainer-2.3.properties*:
https://github.com/OpenLiberty/open-liberty/pull/1971/files#diff-d051bb6d4dfad913af2785e0d18a627a

The properties file is almost identical to the one in jsfContainer-2.2 except 2.2 is replaced with 2.3.

**Comment by hibell** [comment]
#build

File: dev/com.ibm.websphere.appserver.jsfContainer-2.3/resources/l10n/com.ibm.websphere.appserver.jsfContainer-2.3.properties [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#
+#ISMESSAGEFILE FALSE
+#NLS_ENCODING=UNICODE
+#NLS_MESSAGEFORMAT_NONE
+#
+
+description=This feature enables CDI integrations for web applications that use the Java Server Faces \
+ (JSF) 2.3 framework. This feature only includes container integration code for JSF, and does \
+ not include any JSF interfaces or implementation.  To use this feature, JSF interfaces and implementation \
+ must be packaged within web applications that use JSF.
```
Comment (bcpswope) [inline dev/com.ibm.websphere.appserver.jsfContainer-2.3/resources/l10n/com.ibm.websphere.appserver.jsfContainer-2.3.properties]: Looks like some extra spaces here. Other than that, I would name the feature instead of "This feature..." and "needs to be packaged" over "must".

Comment (hibell) [reply dev/com.ibm.websphere.appserver.jsfContainer-2.3/resources/l10n/com.ibm.websphere.appserver.jsfContainer-2.3.properties]: Thanks @bcpswope for the feedback. We want to keep the jsfContainer-2.3.properties file as close to jsfContainer-2.2 version as possible.

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
@pnicolucci The jsfProvider-2.3.0.MyFaces.feature is used in the jsf-2.3 feature file. We can't include it until jsfContainer-2.3 is promoted to beta because the builds will fail due to feature visibility issues. This is tracked in the work item 253228.

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/public/cdi-2.0/com.ibm.websphere.appserver.cdi-2.0.feature [hunk 2]
```diff
@@ -26,7 +27,10 @@ IBM-API-Package: javax.decorator;  type="spec", \
  org.jboss.weld.proxy; type="internal", \
  org.jboss.weld.security; type="internal", \
  org.jboss.weld.serialization.spi; type="internal", \
- org.jboss.weld.context;type="third-party", \
+ org.jboss.weld;type="internal",\
```
Comment (aguibert) [inline dev/com.ibm.websphere.appserver.features/visibility/public/cdi-2.0/com.ibm.websphere.appserver.cdi-2.0.feature]: Why are these needed?  We specifically removed these in cdi-1.2 because of a classloading conflict reported by a user.

Comment (hibell) [reply dev/com.ibm.websphere.appserver.features/visibility/public/cdi-2.0/com.ibm.websphere.appserver.cdi-2.0.feature]: That is needed for the IBMViewHandler to work, otherwise we get CNFEs. I tried to get it to work by exporting to the TCCL but that did not seem to work.

**Comment by hibell** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/public/cdi-2.0/com.ibm.websphere.appserver.cdi-2.0.feature (hunk 2 above)
Comment (aguibert) [reply dev/com.ibm.websphere.appserver.features/visibility/public/cdi-2.0/com.ibm.websphere.appserver.cdi-2.0.feature]: Then we should have the `jsfContainer-2.3` feature expose the Weld packages it needs -- that way the `cdi-2.0` feature doesn't always leak these weld packages to the app classloader.

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
#build

**Comment by hibell** [comment]
#build

File: dev/com.ibm.ws.jsfContainer_fat_2.3/bnd.bnd [hunk 3]
```diff
@@ -0,0 +1,43 @@
+#*******************************************************************************
```
[the remark this answers is not in the document]
Comment (hibell) [reply dev/com.ibm.ws.jsfContainer_fat_2.3/bnd.bnd]: Thanks for the information. I'll try that out sometime.

**Comment by hibell** [comment]
#build
